### PR TITLE
Zenkaku

### PR DESCRIPTION
**Pull request by renierjohn**

### CODE REVIEW

File: src/Plugin/Commerce/PaymentGateway/PaymentASPCommerce_link_type.php [hunk 1]
```diff
@@ -203,43 +201,70 @@ public function onNotify(Request $request) {
...
+	$order 	     = $payment->getOrder();
+	$id 	     = $order->id();
+	$orderData   = $pc->getOrderDetails($order);
+	    $givenName 				   = $order->getData("billing_profile_givenName");
+	    $familyName 			   = $order->getData("billing_profile_familyName");
+	    $payment_gateway_parameter = $order->getData("payment_gateway_parameter");
+
+
+	    $kana_givenName = mb_convert_kana($givenName, "NRS","UTF-8");
+	    $kana_familyName = mb_convert_kana($familyName, "NRS","UTF-8");
```
Comment (renierjohn) [inline src/Plugin/Commerce/PaymentGateway/PaymentASPCommerce_link_type.php:260]: please merge

File: src/Plugin/Commerce/PaymentGateway/PaymentASPCommerce_link_type.php [hunk 2]
```diff
@@ -203,43 +201,70 @@ public function onNotify(Request $request) {
...
+
+// ksm($kana_givenName);
+// ksm($kana_familyName);
+
 	switch ($this->configuration['method_type']) {
 			case 'webcvs':
-				$free_csv = "LAST_NAME=" . $familyName . ",FIRST_NAME=" . $givenName . ",MAIL=" . $order->getEmail(). ",TEL=" . $payment_gateway_parameter;		
+				$free_csv = "LAST_NAME=".$kana_givenName.",FIRST_NAME=".$kana_familyName.",MAIL=" . $order->getEmail(). ",TEL=" . $payment_gateway_parameter . ",BILL_DATE=" . (int)$BILL_DATE;
```
Comment (renierjohn) [inline src/Plugin/Commerce/PaymentGateway/PaymentASPCommerce_link_type.php:267]: please merge

File: src/PluginForm/PaymentASPCommerce_linktype_plugin_form.php [hunk 3]
```diff
@@ -20,33 +38,141 @@ public function buildConfigurationForm(array $form, FormStateInterface $form_sta
...
+
+
+	$postdata = mb_convert_encoding($postdata, 'Shift_JIS', 'UTF-8');
+
+
+	foreach ($postdata as $key => $value) {
+			if (\Drupal::service('payment_asp.languageCheck')->isJapanese($postdata[$key]) || strcmp($key, 'free_csv') == 0) {
+				$postdata[$key] = base64_encode($postdata[$key]);
+		}
+	}
```
Comment (renierjohn) [inline src/PluginForm/PaymentASPCommerce_linktype_plugin_form.php:91]: please merge

Comment (gaaaabf) [reply src/PluginForm/PaymentASPCommerce_linktype_plugin_form.php:91]: sige boss

Comment (gaaaabf) [reply src/PluginForm/PaymentASPCommerce_linktype_plugin_form.php:91]: ok boss

**Comment by gaaaabf** [comment]
Merged manually

**Pull request closed without merging**